pennylane_cirq/simulator_device.py

```diff
@@ -208,12 +211,60 @@ def generate_samples(self):
 
         self._result = self._simulator.run(self.circuit, repetitions=self.shots)
 
-        # Bring measurements to a more managable form, but keep True/False as values for now
-        # They will be changed in the measurement routines where the observable is available
         return np.array(
             [self._result.measurements[str(wire)].flatten() for wire in range(self.num_wires)]
         ).T.astype(int)
 
+    def expval(self, observable, shot_range=None, bin_size=None):
+        # pylint: disable=missing-function-docstring
+        # Analytic mode
+        if self.shots is None:
+            # Observables are in tensor form
+            if isinstance(observable, qml.operation.Tensor):
+                # Projector, Hamiltonian, Hermitian
+                for name in observable.name:
+                    if self._observable_map[name] is None or name == "Projector":
+                        return super().expval(observable, shot_range, bin_size)
+
+                if "Hadamard" in observable.name:
+                    T = qml.operation.Tensor()
+                    for obs in observable.obs:
+                        T = qml.operation.Tensor(T, qml.PauliZ(wires=obs.wires))
+
+                    circuit = self.circuit
+                    obs = cirq.PauliSum() + self.to_paulistring(T)
+                else:
+                    circuit = self.pre_rotated_circuit
+                    obs = cirq.PauliSum() + self.to_paulistring(observable)
+
+                return self._simulator.simulate_expectation_values(
+                    program=circuit,
+                    observables=obs,
+                    initial_state=self._initial_state,
+                )[0].real
```

Comment:
Is unpacking needed here as `simulate_expectation_values` returns a list of expectation values? If so, could it be worth making adjustments later that would support a single call to `simulate_expectation_values` when a QNode returns multiple expectation values?

Reply:
Yes I think it is necessary, the returns of multiple expectation value is done on PennyLane side.